backend/tests/test_integration_dashboard.py

```
@@ -758,3 +758,154 @@ def test_dashboard_profit_take_card_matches_leg_verdict(client, monkeypatch):
+
+
+def test_schwab_pill_downgrades_to_invalid_when_quotes_fail(client, monkeypatch):
+    """When the live quote fan returns ``schwab_failed=True`` the dashboard
+    must downgrade ``status.schwab.valid`` to ``False`` and surface the
+    frozen ``error`` discriminator (issue #277).
+    """
+    from app.services.schwab_client import SchwabClientError
+
+    _patch_status(monkeypatch, schwab_configured=True, fred_key="")
+
+    def fake_get_quote(self, ticker):
```

Ruff (0.15.13)

[warning] Missing return type annotation for private function `fake_get_quote`

Add return type annotation: `NoReturn`

(ANN202)